### PR TITLE
TFT Presets, Generic options, Sanity checks

### PR DESCRIPTION
**Pull request by thinkyhead**
- Make `Configuration.h` TFT options more accessible for configurator tools.
- Add sanity checks for TFT options.
- Set TFT width and height using limited preset options.
- Clean up some older LCD conditionals.
- Updated configuration version to 020008.

### CODE REVIEW

**Review by rhapsodyv — APPROVED** [review]

File: Marlin/Configuration.h [hunk 1]
```diff
@@ -2205,22 +2205,86 @@
...
+  //#define TFT_INTERFACE_FSMC
+  //#define TFT_INTERFACE_SPI
+
+  // TFT Resolution. Enable one of the following options:
+  //#define TFT_RES_320x240
+  //#define TFT_RES_480x272
+  //#define TFT_RES_480x320
+
+  // DOGM upscaling to fit
+  #define GRAPHICAL_TFT_UPSCALE 2   // :[ 2, 3 ]
```
Comment (tpruvot) [inline Marlin/Configuration.h]: this should not be there, upscale can be computed from display resolution

Comment (rhapsodyv) [reply Marlin/Configuration.h]: Indeed. It's already done today in `Marlin/src/lcd/scaled_tft.h`. Good point, it don't need to be exposed to the user!

Comment (thinkyhead) [reply Marlin/Configuration.h]: I thought so too. I believe this can be removed here, and I will look at where it can best be assigned automatically.

Comment (tpruvot) [reply Marlin/Configuration.h]: in fact it is already... its what rhapsody said

**Pull request merged**